test/test_gather.cpp

```diff
@@ -555,4 +584,399 @@ TEST_F(IndexingOpTest, GatherBroadcastInput_CUDA) {
+  fusion.addInput(tv1);
+
+  auto tv2 = add(tv0, IrBuilder::create<Double>(1));
+  auto tv3 = broadcast(tv1, {false, true});
+  auto tv4 = take_along_axis(tv2, tv3, 1);
+  fusion.addOutput(tv4);
+
+  scheduler_utils::prepareForMemoryTypePromotion(&fusion);
+
+  // Test if this split is propagated through the indexed domain
+  tv4->split(1, 10);
+
+  TransformPropagator propagator(tv4);
+  MaxRootDomainInfoSpanningTree(tv4).traverse(&propagator);
+
+  // All of the tensors should have the split by 2, except for tv1.
+  for (auto tv : ir_utils::allTvs(&fusion)) {
+    if (tv == tv1) {
+      continue;
+    }
+    TORCH_CHECK(tv->nDims() == 3, "Unexpected tensor: ", tv->toString());
+    TORCH_CHECK(
+        tv->axis(-1)->definition() &&
+            tv->axis(-1)->definition()->isA<Split>() &&
+            tv->axis(-1)->definition()->as<Split>()->in() ==
+                tv->getMaybeRFactorDomain().at(1),
+        "Unexpected tensor: ",
+        tv->toString());
+  }
+
+  // This should not inline the indexed producer domain. Note that the
+  // producer tensor of the take_along_axis expr is not tv2 as a copy
+  // is inserted
+  inlineMost();
+  auto take_along_axis_input =
+      tv4->definition()->as<TorchGatherOp>()->lookupTv();
+  TORCH_CHECK(
+      take_along_axis_input->getComputeAtPosition() == 1,
+      "Unexpected computeAt position: ",
+      take_along_axis_input->toString());
+
+  // Test if parallelization is propagated through indexed domains
+  tv4->axis(-2)->parallelize(ParallelType::TIDy);
+  tv4->axis(-1)->parallelize(ParallelType::TIDx);
+  scheduler_utils::parallelizeAllLike(tv4);
+
+  for (auto tv : ir_utils::allTvs(&fusion)) {
+    if (tv->isFusionInput()) {
+      continue;
+    }
+    TORCH_CHECK(
+        tv->axis(-2)->getParallelType() == tv4->axis(-2)->getParallelType() &&
+            tv->axis(-1)->getParallelType() == tv4->axis(-1)->getParallelType(),
+        "Unexpected parallelization of tensor: ",
+        tv->toString());
+  }
+
+  // This should make the producer of take_along_axis saved in shared memory
+  scheduler_utils::promoteProducerMemoryTypes(&fusion, {});
+
+  TORCH_CHECK(
+      take_along_axis_input->getMemoryType() == MemoryType::Shared,
+      "Failed to promote memory type: ",
+      take_along_axis_input->toString());
+
+  at::manual_seed(0);
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto options_i = at::TensorOptions().dtype(at::kLong).device(at::kCUDA, 0);
+  auto t0 = at::randn(shape, options);
+  auto t1 = at::randint(0, shape[1], {shape[0]}, options_i);
+  std::vector<c10::IValue> aten_inputs = {t0, t1};
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+
+  auto outputs = fe.runFusion(aten_inputs);
+
+  auto ref = at::take_along_dim(t0 + 1, t1.unsqueeze(-1), 1);
+
+  testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
+}
+
+// Same as the above but with the pointwise scheduler
```

Comment:
Looks like transpose scheduler is also support `take_along_axis`, could you add a test for it?